Tests/PhishingDetectionTests/PhishingDetectionDataActivitiesTest.swift

```diff
@@ -0,0 +1,43 @@
+//
+//  File.swift
+//  
+//
+//  Created by Thom on 04/05/2024.
+//
+
+import Foundation
+import XCTest
+@testable import BrowserServicesKit
+@testable import PhishingDetection
+
+class PhishingDetectionDataActivitiesTest: XCTestCase {
+    var dataActivities: PhishingDetectionDataActivities!
+    var mockDetectionService: MockPhishingDetectionService = MockPhishingDetectionService()
+
+    override func setUp() {
+        super.setUp()
+        dataActivities = PhishingDetectionDataActivities(detectionService: mockDetectionService)
+    }
+
+    func testDataActivitySchedulerRuns() async {
+        let hpExpectation = XCTestExpectation(description: "HashPrefixDataActivity should run and update hash prefixes")
+
+        Task {
+            await dataActivities.run()
+            XCTAssertTrue(mockDetectionService.didUpdateHashPrefixes, "Hash prefixes should be updated")
```

Comment:
You could have both Assertion together after the await XCTWaiter().fulfillment(of: [hpExpectation], timeout: 5.0)